File: ios/Capacitor/CapacitorTests/JSONSerializationWrapper.m

#import "JSONSerializationWrapper.h"

@implementation JSONSerializationWrapper

- (instancetype)initWithDictionary:(NSDictionary *)dictionary {
    self = [super init];
    if (self != nil) {
        _dictionary = dictionary;
    }
    return self;
}

- (NSDictionary *)unwrappedResult {
    NSError* error = nil;
    NSData* serializedData = [NSJSONSerialization dataWithJSONObject:[self dictionary] options:NSJSONWritingPrettyPrinted error:&error];
    if (serializedData != nil) {
        NSDictionary* result = [NSJSONSerialization JSONObjectWithData:serializedData options:0 error:&error];
        return result;
    }
    return nil;
}

@end

File: ios/Tests/CapacitorTests/JSONSerializationWrapper.swift
import Foundation

final class JSONSerializationWrapper {
    let dictionary: [AnyHashable: Any]

    init?(dictionary: [AnyHashable: Any]) {
        self.dictionary = dictionary
    }

    func unwrappedResult() -> [AnyHashable: Any]? {
        guard let serializedData = try? JSONSerialization.data(withJSONObject: dictionary, options: [.prettyPrinted]) else {
            return nil
        }
        return try? JSONSerialization.jsonObject(with: serializedData, options: []) as? [AnyHashable: Any]
    }
}
